contracts/Core/StakeManager.sol

```diff
@@ -393,30 +394,35 @@ contract StakeManager is Initializable, ACL, StakeStorage {
         string memory _reason,
         uint256 _epoch
     ) external onlyRole(parameters.getStakeModifierHash()) {
-        uint256 previousStake = stakers[_id].stake;
-        stakers[_id].stake = _stake;
-        emit StakeChange(
-            _id,
-            previousStake,
-            _stake,
-            _reason,
-            _epoch,
-            block.timestamp
-        );
+        _setStakerStake(_id, _stake, _reason, _epoch);
     }
 
-    /// @notice External function for trasnfering bounty to bountyHunter
-    /// Used by RewardManager in slash()
-    /// @param bountyHunter The address of bountyHunter
-    /// @param bounty The bounty to be rewarded
-    function transferBounty(address bountyHunter, uint256 bounty)
-        external
-        onlyRole(parameters.getStakeModifierHash())
-    {
-        require(
-            razor.transfer(bountyHunter, bounty),
-            "failed to transfer bounty"
-        );
+    /// @notice The function is used by the Votemanager reveal function
+    /// to penalise the staker who lost his secret and make his stake less by "slashPenaltyAmount" and
+    /// transfer to bounty hunter half the "slashPenaltyAmount" of the staker
+    /// @param id The ID of the staker who is penalised
+    /// @param bountyHunter The address of the bounty hunter
+    function slash(
+        uint256 id,
+        address bountyHunter,
+        uint256 epoch
+    ) external onlyRole(parameters.getStakeModifierHash()) {
+        uint256 slashPenaltyAmount = (stakers[id].stake *
+        parameters.slashPenaltyNum())/
+        parameters.slashPenaltyDenom();
+        uint256 newStake =  stakers[id].stake - slashPenaltyAmount;
+        uint256 halfPenalty = slashPenaltyAmount/2;
+
+        if (halfPenalty == 0) return;
+
+        _setStakerStake(id, newStake, "Slashed", epoch);
+        //reward half the amount to bounty hunter
+        //please note that since slashing is a critical part of consensus algorithm,
+        //the following transfers are not `reuquire`d. even if the transfers fail, the slashing
+        //tx should complete.
+        razor.transfer(bountyHunter, halfPenalty);
+        //burn half the amount
+        razor.transfer(parameters.burnAddress(), halfPenalty);
```

Comment:
Hey, Is there a reason 
For choosing this manual transfer for burn

Rather than using one provided by openZap
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/566a774222707e424896c0c390a84dc3c13bdcb2/contracts/token/ERC20/ERC20.sol#L274

If there is, we have to change it for stakedToken as well

https://github.com/razor-network/contracts/blob/808f9289d563a3ac3c3e5306684b4963c5e3f106/contracts/Core/StakeManager.sol#L294

Reply:
With this manual transfer totalSupply remains the same.
With using openZep , totalSupply is decreased by that amount.
so at least in stakedToken we want this behavior, as there we have totalSupply as deciding factor in cal.

Do we want the same for Razor as well? 
IMO we should, as the value of Razor will increase with each burn with it.

Reply:
_burn() is an internal function.
It's not accessible since we did not inherit Burnable.sol for $RAZOR https://github.com/OpenZeppelin/openzeppelin-contracts/blob/566a774222707e424896c0c390a84dc3c13bdcb2/contracts/token/ERC20/extensions/ERC20Burnable.sol

Reply:
Yes, ERC20 have _burn as internal function
We can have one wrapper around it and have it permission based.
Like this maybe
https://github.com/razor-network/contracts/blob/808f9289d563a3ac3c3e5306684b4963c5e3f106/contracts/StakedToken.sol#L16

Reply:
Razor token is already deployed on mainnet. It doesn't have burn function.

Reply:
Oh yes